src/react/components/common/assetPreview/tfrecordAsset.test.tsx

```diff
@@ -42,15 +44,15 @@ describe("Image Asset Component", () => {
         return mount(<TFRecordAsset {...props} />);
     }
 
-    function sleep(ms) {
-        return new Promise((resolve) => setTimeout(resolve, ms));
+    function wait() {
+        return new Promise((resolve) => setImmediate(resolve));
```

Comment:
Does this just wait for all processes to finish? Is this not like just using setImmediate normally?

Reply:
They're not really the same. There're two different queues for io and functions where the second is on top of the other.  setImmediate go directly on the io queue.  From my point of view I reaally don't see a big differences in the browser but wallace suggest me to use setimmediate